OrbitClientGgp/ClientGgp.cpp

```diff
@@ -79,9 +78,105 @@ bool ClientGgp::StopCapture() {
+    LOG("pid:%d, name:%s, path:%s, is64:%d", info.pid(), info.name(), info.full_path(),
+        info.is_64_bit());
+  }
+  auto process_it = find_if(process_infos.begin(), process_infos.end(),
+                            [&pid](const ProcessInfo& info) { return info.pid() == pid; });
+  if (process_it != process_infos.end()) {
+    LOG("Found process by pid, set target process");
+    std::shared_ptr<Process> process = std::make_shared<Process>();
+    process->SetID(process_it->pid());
+    process->SetName(process_it->name());
+    process->SetFullPath(process_it->full_path());
+    process->SetIs64Bit(process_it->is_64_bit());
+    LOG("Process info: pid:%d, name:%s, path:%s, is64:%d", process->GetId(), process->GetName(),
+        process->GetFullPath(), process->GetIs64Bit());
+    return process;
+  }
+  return nullptr;
+}
+
+bool ClientGgp::LoadModuleAndSymbols() {
+  // Load modules for target_process_
+  ErrorMessageOr<std::vector<ModuleInfo>> result_module_infos =
+      process_client_->LoadModuleList(target_process_->GetId());
+  if (result_module_infos.has_error()) {
+    ERROR("Error retrieving modules: %s", result_module_infos.error().message());
+    return false;
+  }
+  // Find module that corresponds to the binary of target_process
+  const std::vector<ModuleInfo>& module_infos = result_module_infos.value();
+  LOG("List of modules");
+  for (const ModuleInfo& info : module_infos) {
+    LOG("name:%s, path:%s, size:%d, address_start:%d. address_end:%d, build_id:%s", info.name(),
+        info.file_path(), info.file_size(), info.address_start(), info.address_end(),
+        info.build_id());
+  }
+  std::string_view target_name = target_process_->GetName();
+  auto module_it =
+      find_if(module_infos.begin(), module_infos.end(),
+              [&target_name](const ModuleInfo& info) { return info.name() == target_name; });
+  if (module_it == module_infos.end()) {
+    ERROR("Error: Module correspondent to process binary not found");
+    return false;
+  }
+  LOG("Found module correspondent to process binary");
+  std::shared_ptr<Module> module = std::make_shared<Module>();
+  module->m_Name = module_it->name();
+  module->m_FullName = module_it->file_path();
+  module->m_PdbSize = module_it->file_size();
+  module->m_AddressStart = module_it->address_start();
+  module->m_AddressEnd = module_it->address_end();
+  module->m_DebugSignature = module_it->build_id();
+  target_process_->AddModule(module);
+  LOG("Module info: name:%s, path:%s, size:%d, address_start:%d. address_end:%d, build_id:%s",
+      module->m_Name, module->m_FullName, module->m_PdbSize, module->m_AddressStart,
+      module->m_AddressEnd, module->m_DebugSignature);
+
+  // Load symbols for the module
+  const std::string& module_path = module->m_FullName;
+  LOG("Looking for debug info file for %s", module_path);
+  ErrorMessageOr<std::string> result_debug_file = process_client_->FindDebugInfoFile(module_path);
```

Comment:
main_executable_debug_file?

Reply:
Makes sense, especially after the other name change, will update, thanks!

Reply:
Done, I also have updated the name of _debug_file_path_ to _main_executable_debug_file_ which is related with this one.